src/cmd-coreos-prune

```diff
@@ -188,20 +190,14 @@ def main():
                                         prune_container(tag, args.dry_run, container_repo, args.registry_auth_file)
                                 else:
                                     print(f"No container tags to prune for build {build_id}.")
+                    action_completion_status[action].append(arch)  # Track success for this action and arch
         # Update policy-cleanup after pruning actions for the architecture
         policy_cleanup = build.setdefault("policy-cleanup", {})
-        for action in policy[stream].keys():  # Only update actions specified in policy[stream]
-            match action:
-                case "cloud-uploads":
-                    if "cloud-uploads" not in policy_cleanup:
-                        policy_cleanup["cloud-uploads"] = True
-                case "images":
-                    if "images" not in policy_cleanup:
-                        policy_cleanup["images"] = True
+        for action, completed_arches in action_completion_status.items():
+            if set(completed_arches) == set(build["arches"]):  # Ensure all arches are completed
```

Comment:
I'm trying to think of a case we would get to this code where not all architectures were completed.. If that were the case wouldn't the code have thrown an exception?

Reply:
That's correct; technically, we could bypass considering the angle of architectures in these changes. However, assuming that pruning is complete for a `build_id` after processing just one architecture felt inconsistent. 
I could revise the implementation to operate that way, and it would technically work.

Reply:
Something like this when we run pruning on the first arch should suffice.
`action_completion_status[action] = True`